src/components/SearchForm/SearchForm.tsx

```diff
@@ -0,0 +1,52 @@
+import React, { useState } from "react";
+import styles from "./SearchForm.module.scss";
+
+const SearchForm: React.FC<{
+  InitialSearchQuery: string;
+  onSearch: (searchQuery: string) => void;
+}> = ({ InitialSearchQuery, onSearch }) => {
+  const [searchQuery, setSearchQuery] = useState(InitialSearchQuery);
+
+  const handleInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setSearchQuery(event.target.value);
+    console.log("handleInputChange", event.target.value);
+  };
+
+  const handleSearchClick = () => {
+    if (onSearch) {
+      onSearch(searchQuery);
+      console.log("handleSearchClick", searchQuery);
+    }
+  };
+
+  const handleKeyPress = (event: React.KeyboardEvent<HTMLInputElement>) => {
```

Comment:
If you use the search button as a submit button, you won't need this handler at all. Pressing "Enter" will work just like button clicks submitting the form, you'll just need to add `event.preventDefault()` to you search click handler.